sdcm/utils/full_storage_utils.py

```diff
@@ -0,0 +1,117 @@
+
+    @staticmethod
+    def calculate_target_used_size(nodes, target_percent):
+        max_total = 0
+        for node in nodes:
+            info = DiskUtils.get_disk_info(node)
+            max_total = max(max_total, info['total'])
+        
+        target_used_size = (target_percent / 100) * max_total
+        current_usage, current_used = DiskUtils.get_max_disk_usage(nodes)
+        additional_usage_needed = target_used_size - current_used
+
+        LOGGER.info(f"Current max disk usage: {current_usage:.2f}%")
+        LOGGER.info(f"Current max used space: {current_used:.2f} GB")
+        LOGGER.info(f"Max total disk space: {max_total:.2f} GB")
+        LOGGER.info(f"Target used space to reach {target_percent}%: {target_used_size:.2f} GB")
+        LOGGER.info(f"Additional space to be used: {additional_usage_needed:.2f} GB")
+
+        return target_used_size
+
+    @staticmethod
+    def log_disk_usage(nodes):
+        for node in nodes:
+            info = DiskUtils.get_disk_info(node)
+            LOGGER.info(f"Disk usage for node {node.name}:")
+            LOGGER.info(f"  Total: {info['total']} GB")
+            LOGGER.info(f"  Used: {info['used']} GB")
+            LOGGER.info(f"  Available: {info['available']} GB")
+            LOGGER.info(f"  Used %: {info['used_percent']}%")
+
+class StressUtils():
+    def __init__(self, db_node, cluster_tester):
+        self.db_node = db_node
+        self.db_cluster = self.db_node.parent_cluster
+        self.cluster_tester = cluster_tester
+        self.total_large_ks = 0
+        self.total_small_ks = 0
+        cores = self.db_cluster.nodes[0].cpu_cores
+        self.num_stress_threads = 8 if not cores else int(cores) * 8
+
+    def prepare_dataset_layout(self, dataset_size, row_size=10240):
+        n = dataset_size * 1024 * 1024 * 1024 // row_size
+        seq_end = n * 100
+
+        return f'cassandra-stress write cl=ONE n={n} -mode cql3 native -rate threads={self.num_stress_threads} ' \
+               f'-pop dist="uniform(1..{seq_end})" ' \
+               f'-col "size=FIXED({row_size}) n=FIXED(1)" ' \
+               f'-schema "replication(strategy=NetworkTopologyStrategy,replication_factor=3)"'
+
+    def run_stress_until_target(self, target_used_size, target_usage):
+        current_usage, current_used = DiskUtils.get_max_disk_usage(self.db_cluster.nodes)
+        smaller_dataset = False
+        
+        space_needed = target_used_size - current_used
+        # Calculate chunk size as 10% of space needed
+        chunk_size = int(space_needed * 0.1)
+        
+        while current_used < target_used_size and current_usage < target_usage:
+            # Write smaller dataset near the threshold (15% or 30GB of the target)
+            smaller_dataset = (((target_used_size - current_used) < 30) or 
+                             ((target_usage - current_usage) <= 15))
+            
+            if not smaller_dataset:
+                self.total_large_ks += 1
```

Comment:
nit: I would prefer large_ks_cnt - in the end it will be total but we use it as a counter

Reply:
ok, renamed total_small_ks as well.